tests/pthread/test_pthread_c11_threads.c

```diff
@@ -17,6 +17,21 @@ void do_once(void) {
   printf("in do_once\n");
 }
 
+// Because this thread is detached it can still be running
+// when the main thread exits.  And becasue of an emscripten
+// bug (https://github.com/emscripten-core/emscripten/issues/15186).
+// this means we can't write to stdout after the main thread
+// exits.  This means we can't use `thread_main` below because
+// the destructor to the `tss_t key` writes to stdout.
+int thread_main_detached(void* arg) {
+  printf("in thread_main_detached %p\n", (void*)thrd_current());
```

Comment:
This writes to stdout here - isn't that as bad as the other print we are avoiding?

Reply:
This happens before the cnd_signal below.. the main thread waits for that signal before exiting so anything before that is fine.